### ISSUE TITLE
omit_local_variable_types - false positive on Null

### ISSUE THREAD

**Issue by nt4f04uNd** [dart-lang/linter#2708]
**Describe the issue**

Enable `omit_local_variable_types`

Write this expression

```dart
const Null response = null;
```

The linter complains that I should omit this type

These workarounds don't work for me:

1. ***The '?' is unnecessary because 'Null' is nullable without it.***

```dart
const Null? response = null;
```

2. ***Missing variable type for 'response'.
Try adding an explicit type, or remove implicit-dynamic from your analysis options file***

```dart
const response = null;
```

I will use `Object?` for now, which works fine, but I would like to limit it to only `Null`

**To Reproduce**
See above

**Expected behavior**
Lint not triggered

**Additional context**
I'm writing tests and all of them should have similar structure - `request` and `response` variables should go at the beginning to be easily discoverable, so I intentionally keep this as a variable

**Comment by eernstg** [comment]
It _does_ make sense for `omit_local_variable_types` to take `Null` into account, because the inferred type for the variable in `var x = e;` where `e` has type `Null` is `dynamic`. So it _does_ make a difference if you actually, really, seriously do want to have a variable of type `Null` once in a blue moon. ;-)
